src/Cli/dotnet/CommandFactory/CommandResolution/MSBuildProject.cs

```diff
@@ -181,8 +182,8 @@ public bool TryGetLockFile(out LockFile lockFile)
                 return false;
             }
 
-            lockFile = new LockFileFormat()
-                .ReadWithLock(lockFilePath)
+            lockFile = Task.Run(() => new LockFileFormat()
+                .ReadWithLock(lockFilePath))
                 .Result;
```

Comment:
Unless there's something about subtle about async in .NET I'm not aware of, I don't understand what this achieves.

the `dotnet` CLI is a console app, and console apps don't have synchronization contexts. Although, this method is using `.Result`, so I'm not sure that synchronization contexts and thread affinity even matters.

Anyway, Task.Run will schedule a task to run on the thread pool. Once a thread pool thread eventually picks up the work, it will execute the `LockFileFormat.ReadWithLock` method, then signal that the task is complete. Then the thread blocked on `.Result` will need to be scheduled by the OS onto a CPU core, and it will continue its work.

I don't understand what theoretical deadlock this might avoid. If all the thread pool threads are blocked on synchronous APIs, it will spin up new threads. Even if console apps had a SynchronizationContext on a single thread, I believe it only avoids deadlocks when the entire callstack uses async. A single `.Result` like this can still deadlock as soon as another methods needs to switch back to the synchronized thread.